---
title: "Figures based on scRNA-seq data"
author: "Friederike DÃ¼ndar, Applied Bioinformatics Core, Weill Cornell Medicine"
date: "updated `r Sys.Date()`"
output:
  pdf_document:
    toc: yes
    toc_depth: '5'
  html_document:
    code_folding: hide
    toc: yes
    toc_depth: 6
editor_options:
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, cache=FALSE, cache.lazy = FALSE)
```

```{r message=FALSE,cache=FALSE, warning=FALSE}
library(magrittr)
library(data.table)
library(scater)
library(ggplot2); theme_set(theme_bw(base_size = 16) )
library(patchwork)
#library(grid)
#library(gridExtra)
```

## UMAP of pancreatic cell types

```{r loading_all_cells, message=FALSE}
## loading data
scf <- readRDS("sce_integratedData_HFD_WT_2019-09.rds")

scf$condition <- ifelse(scf$Sample == "WT", "RD", scf$Sample)
scf$condition <- factor(scf$condition, levels = c("RD","HFD"), ordered = TRUE)

scf$clustLabel <- ifelse(scf$integClust_res0.2 %in% c("0","1","4"), "beta",
  ifelse(scf$integClust_res0.2 == "2", "alpha",
    ifelse(scf$integClust_res0.2 == "3", "delta",
      ifelse(scf$integClust_res0.2 == "5", "endothelial",
        ifelse(scf$integClust_res0.2 == "6", "gamma",
          ifelse(scf$integClust_res0.2 == "7", "immune.cells",
            ifelse(scf$integClust_res0.2 == "8", "epithelial", "stressed/debris")
            ))))))

scf$clustLabel <- factor(scf$clustLabel,
  levels = c("alpha","beta","gamma","delta", "endothelial","epithelial",
    "immune.cells","stressed/debris"), ordered = TRUE)
```

```{r}
## defining color schemes ---------------------------------
hm_col_palette <- colorRampPalette(c("midnightblue","lightslateblue", "lightyellow","yellow","red3"))((7))

cell_colors <- c('alpha'="lightgoldenrod1",'beta'="darkorange1",
  'gamma'="firebrick1",'delta'="firebrick4", 
  'endothelial'= "paleturquoise1",'epithelial'="seagreen3",
  'immune.cells'="plum3",
  'stressed/debris'="gray75")

condition_cols <- c("springgreen2", "orange")
names(condition_cols) <- c("HFD","RD")

cl0.2 <- c('#B1E2F9','limegreen','grey30','#FFC914','#0066E2','#FC71E9','#00E5CA','#E31A1C','grey83','#FF7F00')
names(cl0.2) <- levels(scf$integClust_res0.2)
```

```{r umap_cellTypes, fig.width = 8, fig.height = 7}
scABC2::plot_reducedDim_from_sce(scf,
  which_reddim = "umap_integrated",
  color_by = "clustLabel",
  size_by = .5, alpha = .75,
  add_cell_info = "condition", remove_rug = TRUE,set_color=FALSE) +
  theme_bw() +
  theme(legend.position = "bottom",
    panel.grid.minor=element_blank(),
    panel.grid.major=element_blank()) +
  guides(color =  guide_legend(override.aes = list(alpha = 1, size = 5))) +
  scale_color_manual(values = cell_colors)
```

```{r umap_cellTypes_perCondition, fig.width = 12, fig.height = 7}
scABC2::plot_reducedDim_from_sce(scf,
  which_reddim = "umap_integrated",
  color_by = "clustLabel",
  size_by = .5, alpha = .75,
  add_cell_info = "condition", remove_rug = TRUE,set_color=FALSE) +
  theme_bw() +
  theme(legend.position = "bottom",
    panel.grid.minor=element_blank(),
    panel.grid.major=element_blank()) +
  guides(color =  guide_legend(override.aes = list(alpha = 1, size = 5))) +
  scale_color_manual(values = cell_colors) +
  facet_grid(~condition)
```

## UMAP of beta cell clusters

```{r loading_beta_data}
scf.b <- readRDS("sce_integratedData_betaCells_HFD_WT_2019-09.rds")
scf.b$condition <- ifelse(scf.b$Sample == "WT", "RD", scf.b$Sample)
scf.b$condition <- factor(scf.b$condition, levels = c("RD","HFD"), ordered =TRUE)
scf.b$cluster <- scf.b$clusts0.2

condition_cols <- c('RD'="orange",'HFD'="seagreen2")

cl0.2 <- c("#B1E2F9","limegreen", "grey30","#FFC914")
names(cl0.2) <- levels(scf.b$clusts0.2)
```

```{r umap_beta_cells, fig.width = 8}
scABC2::plot_reducedDim_from_sce(scf.b,
  which_reddim = "umap_beta_integrated",
  color_by = "cluster",
  size_by = .5, alpha = .75,
  add_cell_info = "condition", remove_rug = TRUE,set_color=FALSE) +
  theme_bw() +
  theme(legend.position = "bottom",
    panel.grid.minor=element_blank(),
    panel.grid.major=element_blank()) +
  facet_grid(~condition) +
  scale_color_manual(values = cl0.2) 
```

Number of cells per cluster per condition:

```{r}
table(scf.b$cluster, scf.b$Sample)
```

### Expression of marker genes

The violin plots can easily be reproduced with `dittoSeq::dittoBoxPlots()` instead of our home-grown package.

```{r fig.width = 13, fig.height = 15}
mitogenes <- c("Aldoa","Gapdh", "Atp5g1", "Ndufc2")

mito.dt <- scABC2::make_long_dt(scf.b, features = mitogenes,
  exprs_values = "log1p_sctransform", include_metaData = c("condition","cluster"))

ggplot(mito.dt,
  aes(x = cluster, y = log1p_sctransform)) + 
  geom_violin(aes(color = cluster)) +
  ggbeeswarm::geom_quasirandom(aes(color = cluster), alpha = .5, size = 1) +
  theme(legend.position = "none") +
  ggtitle("Mitochondrial gene expression", subtitle = "Beta cells only") +
  scale_color_manual(values = cl0.2) + 
  facet_grid(feature_name~condition, scales="free_y") +
  xlab("")
```

## CD63 expression

Per cluster

```{r cd63_violins, fig.width = 10, fig.height = 7}
cd63.dt <- scABC2::make_long_dt(scf.b, features = "Cd63",
  include_metaData = c("cluster","condition"),
  exprs_values = "log1p_sctransform")

ggplot(cd63.dt,
  aes(x = cluster, y = log1p_sctransform)) + 
  geom_violin(aes(fill = cluster)) +
  geom_boxplot(fill=NA, width=.1) +
  theme(legend.position = "none") +
  scale_fill_manual(values = cl0.2) +
  facet_grid(feature_name~condition, scales="free_y") +
  xlab("") + #coord_flip() +
  theme(panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank())
```

RD vs HFD

```{r}
ggplot(cd63.dt,
  aes(x = condition, y = log1p_sctransform)) + 
  geom_violin(aes(fill = condition)) +
  geom_boxplot(fill=NA, width=.1) +
  theme(legend.position = "none") +
  scale_fill_manual(values = condition_cols) +
  facet_grid(.~feature_name, scales="free_y") +
  xlab("") + 
  theme(panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank())
```


## Extended Figures

### Expression of pancreatic marker genes 

```{r}
table(scf$Sample, scf$clustLabel)
```


* UMAP of marker gene expression: Ins2, Gcg, Ppy, Sst, Flt1, Cd74


#### RD cells

```{r markers_umap_RDcells, fig.width = 18, fig.height = 13}
drp <- data.frame(reducedDim(scf[, scf$Sample == "WT"], "umap_integrated"))
for(GENE in c("Gcg","Ins2","Ppy","Sst","Flt1","Cd74")){
  drp[[GENE]] <- assay(scf[,rownames(drp)], "log1p_sctransform") %>% .[GENE,, drop = TRUE]
}

drp.l <- reshape2::melt(drp, id.vars = c("UMAP_1","UMAP_2")) %>% as.data.table
drp.l[, rank:= rank(value), by = variable]

pl <- lapply(unique(drp.l$variable), function(GENE){
  P <- ggplot(drp.l[variable == GENE],
    aes(x = UMAP_1, y = UMAP_2, color = value)) + 
    geom_point(alpha = .5, size = .75) + facet_wrap(~variable) +
    scale_color_gradient(low = "grey90", high = "firebrick3") +
    theme(legend.position = "none",
      panel.grid = element_blank())
  return(P)
})
ABCutilities::MultiPlotList(pl, cols = 3)
```

#### HFD cells

```{r markers_umap_HFDcells, fig.width = 18, fig.height = 13}
drp <- data.frame(reducedDim(scf[, scf$Sample == "HFD"], "umap_integrated"))
for(GENE in c("Gcg","Ins2","Ppy","Sst","Flt1","Cd74")){
  drp[[GENE]] <- assay(scf[,rownames(drp)], "log1p_sctransform") %>% 
    .[GENE,, drop = TRUE]
}

drp.l <- reshape2::melt(drp, id.vars = c("UMAP_1","UMAP_2")) %>% as.data.table
drp.l[, rank:= rank(value), by = variable]

pl <- lapply(unique(drp.l$variable), function(GENE){
  P <- ggplot(drp.l[variable == GENE],
    aes(x = UMAP_1, y = UMAP_2, color = value)) + 
    geom_point(alpha = .5, size = .75) + facet_wrap(~variable) +
    scale_color_gradient(low = "grey90", high = "firebrick3") +
    theme(legend.position = "none",
      panel.grid = element_blank())
  return(P)
})
ABCutilities::MultiPlotList(pl, cols = 3)
```

### Diffusion map

```{r}
datadir <- "data/"
sds.beta <- readRDS(file = paste0(datadir, "slingshotResults_betaCells.rds"))
```
```{r}
dittoSeq::dittoDimPlot(scf.b, var = "cluster", reduction.use = "DM_scaled", 
    add.trajectory.curves = sds.beta@curves, main = "DCs 1-2 as determined with slingshot",
    color.panel = cl0.2,
    legend.title = "cluster") +
    theme(legend.position = "bottom")
```

### Cd63 in different pancreatic cell types

* all cells, by condition and cell label

```{r fig.width = 10}
cd63.dt <- scABC2::make_long_dt(scf, features = "Cd63",
  include_metaData = c("clustLabel","condition"),
  exprs_values = "log1p_sctransform")

ggplot(cd63.dt,
  aes(x = clustLabel, y = log1p_sctransform)) + 
  geom_violin(aes(fill = clustLabel)) +
  geom_boxplot(fill=NA, width=.1) +
#  ggbeeswarm::geom_quasirandom(aes(color = clustLabel), alpha = .25, size = 1) +
  theme(legend.position = "none") +
  ggtitle("Cd63 expression per cell type") +
 # scale_color_manual(values = cell_colors) + 
  scale_fill_manual(values = cell_colors) +
  facet_grid(feature_name~condition, scales="free_y") +
  xlab("") + #coord_flip() +
  theme(panel.grid.major.x = element_blank(), panel.grid.minor.x = element_blank(),
    axis.text.x  = element_text(angle=90, vjust=0.5, size=14))
```



## SessionInfo

```
## R version 3.6.2 (2019-12-12)
## Platform: x86_64-apple-darwin15.6.0 (64-bit)
## Running under: macOS Catalina 10.15.4 ##
## Matrix products: default

## BLAS: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib ##
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8 ##
## attached base packages:
## [1] stats4 parallel stats
## [8] methods   base
##
## other attached packages:
##  [1] ReactomePA_1.30.0           clusterProfiler_3.14.3     
##  [3] destiny_3.0.1               dittoSeq_1.1.2             
##  [5] slingshot_1.4.0             princurve_2.1.4            
##  [7] patchwork_1.0.0             scater_1.14.6              
##  [9] ggplot2_3.3.0               SingleCellExperiment_1.8.0 
## [11] SummarizedExperiment_1.16.1 DelayedArray_0.12.3        
## [13] BiocParallel_1.20.1         matrixStats_0.56.0         
## [15] Biobase_2.46.0              GenomicRanges_1.38.0       
## [17] GenomeInfoDb_1.22.1         IRanges_2.20.2             
## [19] S4Vectors_0.24.4            BiocGenerics_0.32.0        
## [21] data.table_1.12.8           magrittr_1.5               
## 
## loaded via a namespace (and not attached):
##   [1] rappdirs_0.3.1           ggthemes_4.2.0           tidyr_1.0.2             
##   [4] bit64_0.9-7              knitr_1.28               irlba_2.3.3             
##   [7] RCurl_1.98-1.1           cowplot_1.0.0            RSQLite_2.2.0           
##  [10] RANN_2.6.1               europepmc_0.3            proxy_0.4-24            
##  [13] future_1.16.0            bit_1.1-15.2             enrichplot_1.6.1        
##  [16] webshot_0.5.2            xml2_1.3.1               assertthat_0.2.1        
##  [19] viridis_0.5.1            xfun_0.13                hms_0.5.3               
##  [22] evaluate_0.14            DEoptimR_1.0-8           fansi_0.4.1             
##  [25] progress_1.2.2           caTools_1.18.0           readxl_1.3.1            
##  [28] htmlwidgets_1.5.1        igraph_1.2.5             DBI_1.1.0               
##  [31] purrr_0.3.3              ellipsis_0.3.0           RSpectra_0.16-0         
##  [34] dplyr_0.8.5              backports_1.1.6          vctrs_0.2.4             
##  [37] TTR_0.23-6               ROCR_1.0-7               abind_1.4-5             
##  [40] RcppEigen_0.3.3.7.0      withr_2.1.2              ggforce_0.3.1           
##  [43] triebeard_0.3.0          robustbase_0.93-6        checkmate_2.0.0         
##  [46] vcd_1.4-7                sctransform_0.2.1        scran_1.14.6            
##  [49] xts_0.12-0               prettyunits_1.1.1        cluster_2.1.0           
##  [52] DOSE_3.12.0              lazyeval_0.2.2           ape_5.3                 
##  [55] laeken_0.5.1             crayon_1.3.4             labeling_0.3            
##  [58] edgeR_3.28.1             pkgconfig_2.0.3          tweenr_1.0.1            
##  [61] nlme_3.1-147             vipor_0.4.5              drat_0.1.5              
##  [64] nnet_7.3-13              rlang_0.4.5              globals_0.12.5          
##  [67] lifecycle_0.2.0          rsvd_1.0.3               cellranger_1.1.0        
##  [70] polyclip_1.10-0          RcppHNSW_0.2.0           lmtest_0.9-37           
##  [73] graph_1.64.0             Matrix_1.2-18            urltools_1.7.3          
##  [76] carData_3.0-4            boot_1.3-24              zoo_1.8-7               
##  [79] beeswarm_0.2.3           ggridges_0.5.2           pheatmap_1.0.12         
##  [82] png_0.1-7                viridisLite_0.3.0        bitops_1.0-6            
##  [85] KernSmooth_2.23-16       blob_1.2.1               DelayedMatrixStats_1.8.0
##  [88] stringr_1.4.0            qvalue_2.18.0            readr_1.3.1             
##  [91] gridGraphics_0.5-0       reactome.db_1.70.0       scales_1.1.0            
##  [94] memoise_1.1.0            graphite_1.32.0          plyr_1.8.6              
##  [97] hexbin_1.28.1            ica_1.0-2                gplots_3.0.3            
## [100] gdata_2.18.0             zlibbioc_1.32.0          compiler_3.6.2          
## [103] lsei_1.2-0               ABCutilities_0.3.3       dqrng_0.2.1             
## [106] kableExtra_1.1.0         RColorBrewer_1.1-2       pcaMethods_1.78.0       
## [109] fitdistrplus_1.0-14      cli_2.0.2                XVector_0.26.0          
## [112] listenv_0.8.0            pbapply_1.4-2            ggplot.multistats_1.0.0 
## [115] MASS_7.3-51.5            tidyselect_1.0.0         stringi_1.4.6           
## [118] forcats_0.5.0            highr_0.8                yaml_2.2.1              
## [121] GOSemSim_2.12.1          BiocSingular_1.2.2       locfit_1.5-9.4          
## [124] ggrepel_0.8.2            grid_3.6.2               fastmatch_1.1-0         
## [127] tools_3.6.2              future.apply_1.5.0       rio_0.5.16              
## [130] rstudioapi_0.11          foreign_0.8-76           gridExtra_2.3           
## [133] smoother_1.1             scatterplot3d_0.3-41     farver_2.0.3            
## [136] Rtsne_0.15               ggraph_2.0.2             digest_0.6.25           
## [139] rvcheck_0.1.8            BiocManager_1.30.10      Rcpp_1.0.4              
## [142] car_3.0-8                RcppAnnoy_0.0.16         httr_1.4.1              
## [145] AnnotationDbi_1.48.0     npsurv_0.4-0             scABC2_0.3.1            
## [148] colorspace_1.4-1         reticulate_1.15          rvest_0.3.5             
## [151] ranger_0.12.1            splines_3.6.2            statmod_1.4.34          
## [154] uwot_0.1.8               graphlayouts_0.6.0       sp_1.4-1                
## [157] ggplotify_0.0.5          plotly_4.9.2.1           jsonlite_1.6.1          
## [160] tidygraph_1.1.2          R6_2.4.1                 pillar_1.4.3            
## [163] htmltools_0.4.0          glue_1.4.0               VIM_5.1.1               
## [166] BiocNeighbors_1.4.2      class_7.3-16             codetools_0.2-16        
## [169] fgsea_1.12.0             tsne_0.1-3               lattice_0.20-41         
## [172] tibble_3.0.0             curl_4.3                 ggbeeswarm_0.6.0        
## [175] leiden_0.3.3             gtools_3.8.2             zip_2.0.4               
## [178] GO.db_3.10.0             openxlsx_4.1.4           survival_3.1-12         
## [181] limma_3.42.2             rmarkdown_2.1            munsell_0.5.0           
## [184] e1071_1.7-3              DO.db_2.9                GenomeInfoDbData_1.2.2  
## [187] haven_2.2.0              reshape2_1.4.4           gtable_0.3.0            
## [190] Seurat_3.1.5
```
